pkg/store/pipelineconfig/store.go

```diff
@@ -0,0 +1,238 @@
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package pipelineconfig
+
+import (
+	"context"
+	atomixclient "github.com/atomix/go-client/pkg/client"
+	"github.com/atomix/go-client/pkg/generic"
+	"github.com/atomix/go-client/pkg/primitive"
+	atomicmap "github.com/atomix/go-client/pkg/primitive/atomic/map"
+	p4rtapi "github.com/onosproject/onos-api/go/onos/p4rt/v1"
+	"github.com/onosproject/onos-lib-go/pkg/errors"
+	"github.com/onosproject/onos-lib-go/pkg/logging"
+	"io"
+	"time"
+)
+
+var log = logging.GetLogger()
+
+// Store P4 pipeline pipelineconfig store interface
+type Store interface {
+	// Get gets the pipelineconfig intended for a given target ID
+	Get(ctx context.Context, id p4rtapi.PipelineConfigID) (*p4rtapi.PipelineConfig, error)
+
+	// Create creates a p4 pipeline pipeline config
+	Create(ctx context.Context, pipelineConfig *p4rtapi.PipelineConfig) error
+
+	// Update updates a p4 pipeline pipeline config
+	Update(ctx context.Context, pipelineConfig *p4rtapi.PipelineConfig) error
+
+	// List lists all the pipeline config
+	List(ctx context.Context) ([]*p4rtapi.PipelineConfig, error)
+
+	// Watch watches pipeline config changes
+	Watch(ctx context.Context, ch chan<- *p4rtapi.PipelineConfig, opts ...WatchOption) error
+
+	// UpdateStatus updates a pipeline config status
+	UpdateStatus(ctx context.Context, pipelineConfig *p4rtapi.PipelineConfig) error
+
+	Close(ctx context.Context) error
+}
+
+type watchOptions struct {
+	configurationID p4rtapi.PipelineConfigID
+	replay          bool
+}
+
+// WatchOption is a pipelineconfig option for Watch calls
+type WatchOption interface {
+	apply(*watchOptions)
+}
+
+// watchReplyOption is an option to replay events on watch
+type watchReplayOption struct {
+}
+
+func (o watchReplayOption) apply(options *watchOptions) {
+	options.replay = true
+}
+
+// WithReplay returns a WatchOption that replays past changes
+func WithReplay() WatchOption {
+	return watchReplayOption{}
+}
+
+type watchIDOption struct {
+	id p4rtapi.PipelineConfigID
+}
+
+func (o watchIDOption) apply(options *watchOptions) {
+	options.configurationID = o.id
+}
+
+// WithPipelineConfigID returns a Watch option that watches for configurations based on a given pipeline pipelineconfig ID
+func WithPipelineConfigID(id p4rtapi.PipelineConfigID) WatchOption {
+	return watchIDOption{id: id}
+}
+
+type configurationStore struct {
+	pipelineConfigs atomicmap.Map[p4rtapi.PipelineConfigID, *p4rtapi.PipelineConfig]
+}
+
+// NewAtomixStore creates a new Atomix store for device pipeline configurations
+func NewAtomixStore(client primitive.Client) (Store, error) {
+	pipelineConfigsAtomicMap, err := atomixclient.AtomicMap[p4rtapi.PipelineConfigID, *p4rtapi.PipelineConfig](client)("device-provisioner-pipeline-configurations").
+		Codec(generic.GoGoProto[*p4rtapi.PipelineConfig](&p4rtapi.PipelineConfig{})).
+		Get(context.Background())
+
+	if err != nil {
+		return nil, errors.FromAtomix(err)
```

Comment:
I think we're going to need a new implementation of this `FromAtomix` function. This is for converting errors from the _old_ client, but you're going to be giving it errors from the _new_ client.